pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -2043,7 +2043,7 @@ def _is_node_return_ended(self, node: nodes.NodeNG) -> bool:
                 return any(
                     self._is_node_return_ended(_child) for _child in all_but_handler
                 ) and all(self._is_node_return_ended(_child) for _child in handlers)
-            case nodes.Assert(test=nodes.Const(value=value)) if not value:
+            case nodes.Assert(test=nodes.Const(value=False | 0)):
                 # consider assert False as a return node
                 return True
```

Comment:
Guess it's not enough to handle just `assert False`. Pytest also uses `assert 0`. Technically `assert None` or `assert ""` and similar could also be used, though I don't think they are common enough.